### PR TITLE
feat(@snyk/fix): issues breakdown by severity

### PR DESCRIPTION
**Pull request by lili2311**
- [x] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/.github/CONTRIBUTING.md) rules

#### What does this PR do?
- Show a summary of total issues + breakdown by severity
- Count how many are fixable via patch or upgrade from
remediation data for now as only projects with remediation data are supported.

#### Screenshots
CleanShot 2021-04-15 at 09 58 56@2x

#### Additional
Note `Fixed` issues count is coming in a next PR

### CODE REVIEW

File: packages/snyk-fix/src/lib/issues/fixable-issues.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  results: TestResult[],
+): {
+  hasFixes: boolean;
+  count: number;
+} {
+  let hasFixes = false;
+  let count = 0;
+  for (const result of Object.values(results)) {
+    const { remediation } = result;
+    if (remediation) {
```
Comment (lili2311) [inline packages/snyk-fix/src/lib/issues/fixable-issues.ts:13]: using remediation data for now for the counts, when we can send new Issues in full we can migrate to use that.

File: packages/snyk-fix/src/lib/issues/fixable-issues.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      const { upgrade, pin, patch } = remediation;
+      const upgrades = Object.keys(upgrade);
+      const pins = Object.keys(pin);
+      if (pins.length || upgrades.length) {
+        hasFixes = true;
+        // pins & upgrades are mutually exclusive
+        count += getUpgradableIssues(pins.length ? pin : upgrade);
+      }
+      if (Object.keys(patch).length) {
```
Comment (aarlaud) [inline packages/snyk-fix/src/lib/issues/fixable-issues.ts:22]: shouldn't we do the same as for pins and upgrades here? const patches = Object.keys(patch) etc so it's consistent?

File: packages/snyk-fix/src/types.ts [hunk 3]
```diff
@@ -56,8 +56,8 @@ interface UpgradePath {
 }
 
 export interface FixInfo {
-  upgradePaths: UpgradePath[];
-  isPatchable: boolean;
+  upgradePaths?: UpgradePath[];
+  isPatchable?: boolean;
```
Comment (aarlaud) [inline packages/snyk-fix/src/types.ts:60]: do we have isPinnable to?

**Review by aarlaud — APPROVED** [review]

File: packages/snyk-fix/src/types.ts (hunk 3 above)
Comment (lili2311) [reply packages/snyk-fix/src/types.ts:60]: it is not yet there, see slack

File: packages/snyk-fix/src/lib/issues/fixable-issues.ts (hunk 2 above)
Comment (lili2311) [reply packages/snyk-fix/src/lib/issues/fixable-issues.ts:22]: good catch

**Pull request merged**